### PR TITLE
Compile client on Dotty

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

File: client/src/main/scala/org/http4s/client/middleware/FollowRedirect.scala [hunk 1]
```diff
@@ -98,7 +98,8 @@ object FollowRedirect {
         case Right((resp, dispose)) =>
           (methodForRedirect(req, resp), resp.headers.get(Location)) match {
             case (Some(method), Some(loc)) if redirects < maxRedirects =>
-              val nextReq = nextRequest(req, loc.uri, method, resp.cookies)
+              // Dotty needs the asInstanceOf help.
+              val nextReq = nextRequest(req, (loc.asInstanceOf[Location]).uri, method, resp.cookies)
```
Comment (rossabaker) [inline client/src/main/scala/org/http4s/client/middleware/FollowRedirect.scala:102]: This is where the `HeaderT` projection starts to raise its ugly head.

**Review by rossabaker — COMMENTED** [review]
I held back 1334783 from 0.21 because I think they sludge up the code and hoped someone has a better idea.

File: client/src/main/scala/org/http4s/client/middleware/FollowRedirect.scala (hunk 1 above)
Comment (hamnis) [reply client/src/main/scala/org/http4s/client/middleware/FollowRedirect.scala:102]: indeed. but you can avoid that by assigning 
`resp.headers.get(Location)` to a val

its still not good, but avoids asInstanceOf

Comment (rossabaker) [reply client/src/main/scala/org/http4s/client/middleware/FollowRedirect.scala:102]: Oh, nice. I like that better.

**Comment by rossabaker** [comment]
Most of this is on the target branch already. Will start clean on a new PR.

**Pull request closed without merging**